src/ProjectSystemTools/TableControl/TableSearchFilter.cs

```diff
@@ -29,10 +30,17 @@ public TableSearchFilter(IVsSearchQuery searchQuery, IWpfTableControl control)
         public bool Match(ITableEntryHandle entry)
         {
             var cachedColumnValues = new string[_visibleColumns.Count + 1];
-
-            return _searchTokens.Where(searchToken => !(searchToken is IVsSearchFilterToken))
-                .All(searchToken => AtLeastOneColumnOrDetailsContentMatches(entry, searchToken,
-                    cachedColumnValues));
+            ThreadHelper.ThrowIfNotOnUIThread();
+            return _searchTokens.Where(searchToken =>
+            {
+                ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
I would expect this case to be covered by the test on line 33. Is an analyzer flagging this?